hedera-node/cli-clients/src/main/java/com/hedera/services/cli/signedstate/DumpContractBytecodesSubcommand.java

```diff
@@ -0,0 +1,230 @@
+    final EmitSummary emitSummary;
+    final Uniqify uniqify;
+    final WithIds withIds;
+    final Verbosity verbosity;
+
+    DumpContractBytecodesSubcommand(
+            @NonNull final SignedStateHolder state,
+            @NonNull final Path bytecodePath,
+            @NonNull final EmitSummary emitSummary,
+            @NonNull final Uniqify uniqify,
+            @NonNull final WithIds withIds,
+            @NonNull final Verbosity verbosity) {
+        this.state = state;
+        this.bytecodePath = bytecodePath;
+        this.emitSummary = emitSummary;
+        this.uniqify = uniqify;
+        this.withIds = withIds;
+        this.verbosity = verbosity;
+    }
+
+    void doit() {
+
+        var r = getNonTrivialContracts();
+        var contractsWithBytecode = r.getLeft();
+        var zeroLengthContracts = r.getRight();
+
+        final var totalContractsRegisteredWithAccounts = contractsWithBytecode.registeredContractsCount();
+        final var totalContractsPresentInFileStore =
+                contractsWithBytecode.contracts().size();
+        int totalUniqueContractsPresentInFileStore = totalContractsPresentInFileStore;
+
+        if (uniqify == DumpStateCommand.Uniqify.YES) {
+            r = uniqifyContracts(contractsWithBytecode, zeroLengthContracts);
+            contractsWithBytecode = r.getLeft();
+            zeroLengthContracts = r.getRight();
+            totalUniqueContractsPresentInFileStore =
+                    contractsWithBytecode.contracts().size();
+        }
+
+        final var sb = new StringBuilder(estimateReportSize(contractsWithBytecode));
+        if (emitSummary == DumpStateCommand.EmitSummary.YES) {
+            sb.append("%d registered contracts, %d with bytecode (%d are zero-length)%s, %d deleted contracts%n"
+                    .formatted(
+                            totalContractsRegisteredWithAccounts,
+                            totalContractsPresentInFileStore + zeroLengthContracts.size(),
+                            zeroLengthContracts.size(),
+                            uniqify == DumpStateCommand.Uniqify.YES
+                                    ? ", %d unique (by bytecode)".formatted(totalUniqueContractsPresentInFileStore)
+                                    : "",
+                            contractsWithBytecode.deletedContracts().size()));
+        }
+        appendFormattedContractLines(sb, contractsWithBytecode);
+
+        if (verbosity == Verbosity.VERBOSE) System.out.printf("=== Have %d byte report%n", sb.length());
+
+        writeReportToFile(sb.toString());
+    }
+
+    int estimateReportSize(@NonNull Contracts contracts) {
+        int totalBytecodeSize = contracts.contracts().stream()
+                .map(Contract::bytecode)
+                .mapToInt(bc -> bc.length)
+                .sum();
+        // Make a swag based on how many contracts there are plus bytecode size
+        int reportSizeEstimate = contracts.registeredContractsCount() * 75 + totalBytecodeSize * 2;
```

Comment:
Can you help me understand the reasoning for this formula?  Maybe a constant for the 75 might help clarify?

Reply:
ok! it's the word "SWAG" that's a hint.  Basically, I'm allowing enough in the text file for the bytecode (that's the `*2` to account for bytes->hex digits, and there's one line per contract and in addition to the bytecode you also get a list of all the contract ids for that bytecode (if it's been deduped) so I'm just estimating that that's maybe 75 additional characters per line _on average_.  It's pretty rough, and an over estimate, and I just wanted to avoid reallocating the string buffer, and it's certainly totally unnecessary for a command line tool (i.e., not production) but there it is.

Reply:
I'll put a comment in _later_ so I don't lose approvals on this one ... (I won't forget)

Reply:
done